### DISCUSSION TITLE
Fleshing out contents of microlabs

### DISCUSSION THREAD

**Discussion by chaosvolt** [cataclysmbn/Cataclysm-BN#2050] [General]
# Is your feature request related to a problem? Please describe.

Compared to regular labs, microlabs are noticeably flawed. Their monster spawns lean towards being significantly more dangerous than regular labs, and for what? For the most part, nothing but non-classified chemical items, 90% of which do nothing. Luckily Lamandus has been working to fix that last bit via giving these chemicals actual utility, but I did have some ideas as well for further improvements for microlabs.

The main, or really only, functional niche microlabs have is the ample supply of chemicals and chemistry tools, which can hypothetically come in handy later one when you're exploring standard labs. A good direction to take would thus be expanding the variety of what types of rooms show up in microlabs with a specific eye towards being a supplementary location. Basically, materials that are less directly useful on their own but extremely useful for getting into late-game science, nudging the player to explore both types of labs.

# Describe the solution you'd like

What came to mind was a series of ideas for possible room variations and probable contents:
1. Chems and chemistry tools. This is what's currently basically the only thing you find in minilabs. Could be improved by potentially having some rooms spawn a chemistry cart.
2. Electronics supplies. There could be rooms have have solder, wire, other basic components and tools useful for crafting. A battery charging cart could be a useful thing to see in some of these. Spawns of batteries or even plutonium cells would be a good rare spawn to add.
3. Medical equipment, including stuff related to the operation of both autodocs and centrifuges. Should be a potential source of blood draw kits, scalpels, autoclave pouches, anesthetic and/or anesthesia kits, and/or the basic chemicals needed to craft anesthesia. If minilabs don't already include a constructed autoclave, then a rare variant could include either a guaranteed spawn of its item form, a constructed autoclave, or a vehicle autoclave on a cart.
4. Nanomaterial vault. This should be uncommon to rare, and have decent security. Obviously, it should primarily store nanomaterial. Could maybe also have a chance of spawning an STC or two?
5. CVD supplies. Another location that should ideally be rare with decent security. Main purpose is secure, fireproof storage of hydrogen canisters, and potentially include coal or charcoal. Given regular chems storage will likely have charcoal and that's craftable anyway, it could perhaps just be storage for hydrogen canisters, maybe also hydrogen tanks for flavor.
6. Welding supply workshop. Could have welders and related materials, but main focus would be the same idea as the workshop in the industrial center, being a place to spawn tools and materials required to produce oxy-acetylene for projects that need it.
7. Plutonium storage. If plutonium cells from point two are kept separately from regular battery storage, then this would be a good place for it. Should be a decently sealed-off room with a decontamination room and hazmat equipment perhaps stored nearby. Inside would be plutonium cells, plutonium batteries, maybe even a chance of military cells needed by mechs. Irradiating it would introduce some added risk/reward to it.

My thinking here is these would be mixed in with the regular generic chem storage rooms as alternative weights for the standard `microlab_generic` mapgen, possibly `microlab_generic_edge` depending on contents of those.

My rough idea for the spawn weights would probably be something along the lines of about 50% chemical storage, with the variants occupying varying amounts of the other 50%. Nanomaterial, hydrogen, and plutonium storage probably should not exceed 5-10% each, unsure what exact weights would be best.

# Describe alternatives you've considered

1. Adding additional non-storage rooms that either replicate the function of normal labs (example, evidently there can be autoclave rooms in microlabs already but these seem to be vanishingly rare as usual) or add something unique entirely (example: the dimensional shifting hallway, also something I've yet to actually encounter and admittedly a bad example as it's purely a novelty).
2. Making sure all the chemical ingredients and tools are useful. This is going to be a fair bit of effort since DDA's chemistry overhaul seems to be abandoned, and chances are some of the stuff is still going to be useless by the end of it and in need of obsoletion.

Of course, no reason not to do all of the above, making microlabs more dynamic and varied while also ensuring that the bog-standard spawns are also not completely useless.

# Additional context

I'd need to check how many interesting things actually spawn in microlabs and what their weights are. So far I've yet to encounter anything but chem rooms so whatever's there seems vanishingly rare.

**Comment by Lamandus** [comment]
> 2. Making sure all the chemical ingredients and tools are useless.

is that REALLY what you consider as an alternative, volt?

**Comment by chaosvolt** [comment]
Pffft, typo on my part.
